src/igr-tests/igr-runner.cc

```diff
@@ -998,3 +999,43 @@ void svc_arg_test()
             "foo\n" +
             "bar\n");
 }
+
+void svc_arg_consumer_test()
+{
+    igr_test_setup setup("svc-arg-consumer");
```

Comment:
Could you add some comments throughout this test explaining (broadly) what is happening at each stage.

See for example `add_rm_dep_test` and `before_after2_test` for examples of tests with good comments.